wp-contributions.php

```diff
@@ -176,6 +182,70 @@ public function display_card( $args = array() ) {
 			return $card;
 
 		}
+
+		/**
+		 * Block 2 Initializer.
+		 */
+		public function wp_contributions_block_init() {
+			register_block_type(
+				plugin_dir_path( __FILE__ ) . 'assets/block/build',
+				array(
+					'render_callback' => 'wp_contributions_block_callback',
+					'attributes'      => array(
+						'slug' => array(
+							'type' => 'string',
+						),
+						'preferred_username' => array(
+							'type'    => 'string',
+							'default' => 'My Preferred_handle',
+						),
+						'theme'  => array(
+							'type'    => 'boolean',
+							'default' => false,
+						),
+						'contribution_type' => array(
+							'type'    => 'string',
+							'default' => 'plugin',
+						),
+					),
+				)
+			);
+		}
+	}
+
+	/**
+	 * Block Output.
+	 *
+	 * @param string $attr Block attributes
+	 * @return string The HTML output for the card view.
+	 */
+	function wp_contributions_block_callback( $attr ) {
+		error_log('what? block callback! ');
+		extract( $attr );
```

Comment:
I know it feels like a convenient function, but it's my understanding that `extract` can actually be a bad thing for developers and lead to security issues among some other things. I can find examples if really desired, but for now lets stick with `$attr['slug']` and similar and just access from the array itself

Reply:
I will find a replacing function or just look by keys into the array, good to know